packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -196,11 +176,44 @@ contract PluginRepo is
         Tag memory tag = Tag(_release, build);
         bytes32 _tagHash = tagHash(tag);
 
-        versions[_tagHash] = Version(tag, _pluginSetup, _contentURI);
+        versions[_tagHash] = Version(tag, _pluginSetup, _buildMetadata);
 
         latestTagHashForPluginSetup[_pluginSetup] = _tagHash;
 
-        emit VersionCreated(_release, build, _pluginSetup, _contentURI);
+        emit VersionCreated(_release, build, _pluginSetup, _buildMetadata);
+
+        if (_releaseMetadata.length > 0) {
+            _updateReleaseMetadata(_release, _releaseMetadata);
+        }
+    }
+
+    /// @inheritdoc IPluginRepo
+    function updateReleaseMetadata(
+        uint8 _release,
+        bytes calldata _metadata
+    ) external auth(address(this), MAINTAINER_PERMISSION_ID) {
+        if (_release == 0) {
+            revert ReleaseZeroNotAllowed();
+        }
+
+        if (_release > latestRelease) {
+            revert ReleaseDoesNotExist({release: _release});
+        }
+
+        if (_metadata.length == 0) {
+            revert ReleaseMetadataInvalid({release: _release, metadata: _metadata});
+        }
+
+        _updateReleaseMetadata(_release, _metadata);
+    }
+
+    /// @notice The private helper function to replace new `_metadata` for the `_release`.
+    /// @param _release The release number.
+    /// @param _metadata External URI where the plugin's release metadata and subsequent resources can be fetched from.
+    function _updateReleaseMetadata(uint8 _release, bytes calldata _metadata) internal {
+        metadataPerRelease[_release] = _metadata;
```

Comment:
Could it make sense to only allow updating the metadata of the current release and not the older?

Not important, resolve if unsure :+1: